nipoppy/workflows/dataset_init.py

```diff
@@ -37,20 +41,38 @@ def __init__(
             dry_run=dry_run,
         )
         self.fname_readme = "README.md"
+        self.bids_source = bids_source
 
     def run_main(self):
-        """Create dataset directory structure."""
+        """Create dataset directory structure.
+
+        Create directories and add a readme in each.
+        Copy boutiques descriptors and invocations.
+        Copy default config files.
+
+        If the BIDS source dataset is requested, it is copied.
+        """
         # dataset must not already exist
         if self.dpath_root.exists():
             raise FileExistsError("Dataset directory already exists")
 
         # create directories
         for dpath in self.layout.dpaths:
-            self.mkdir(dpath)
+
+            # If a bids_source is passed it means datalad is installed.
+            if self.bids_source is not None and dpath.stem == "bids":
+                self.logger.info(
+                    f"Installing datalad BIDS raw dataset from {self.bids_source}."
+                )
+                copytree(self.bids_source, str(dpath))
```

Comment:
note that no BIDS validation is done before copying the data

Reply:
also this copies the data instead of symlink or moving it: to be discussed

Reply:
Having it copy by default makes sense since that is the safest option, but it would be good to have the option to create a symlink or move it just in case the BIDS dataset is huge

Reply:
yeah: maybe let's keep it like this for now and maybe we can think of adding another CLI argument to say:
 
--mode: copy
--mode: move
--mode: link
--mode: datalad